crypto/batchverifier.go

```diff
@@ -164,5 +172,10 @@ func batchVerificationImpl(messages [][]byte, publicKeys []SignatureVerifier, si
 		C.size_t(len(messages)),
 		(*C.int)(unsafe.Pointer(valid)))
 
-	return allValid == 0
+	failed = make([]bool, numberOfSignatures)
+	for i := 0; i < numberOfSignatures; i++ {
+		cint := *(*C.int)(unsafe.Pointer(uintptr(valid) + uintptr(i*C.sizeof_int)))
+		failed[i] = (cint == 0)
```

Comment:
nits: Why not take the positive approach? i.e index `i` is set to `true` if the `i` signature is valid? (nothing wrong with the current approach - just wounder if that would be clearer )